File: src/org/cerion/symcalc/expression/function/list/Last.java
package org.cerion.symcalc.expression.function.list;

import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.FunctionExpr;
import org.cerion.symcalc.expression.ListExpr;

public class Last extends FunctionExpr {

	public Last(Expr... e) {
		super(FunctionType.LAST,e);
	}
	
	@Override
	protected Expr evaluate() {
		
		if(get(0).isList())
		{
			ListExpr L = (ListExpr)get(0);
			return L.get(L.size() - 1);
		}
		
		return this;
	}
}

File: src/org/cerion/symcalc/expression/function/list/Last.kt
package org.cerion.symcalc.expression.function.list

import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.expression.FunctionExpr

class Last(vararg e: Expr) : FunctionExpr(FunctionExpr.FunctionType.LAST, *e) {

    override fun evaluate(): Expr {
        if (get(0).isList) {
            val l = get(0).asList()
            return l[l.size() - 1]
        }

        return this
    }
}
